lib/chef/resource/aws_subnet.rb

```diff
@@ -75,6 +76,12 @@ class Chef::Resource::AwsSubnet < Chef::Provisioning::AWSDriver::AWSResourceWith
   #
   attribute :route_table#, kind_of: [ String, AwsRouteTable, AWS::EC2::RouteTable ], equal_to: [ :default_to_main ]
 
+  #
+  # The Network ACL to associate with this subnet. Subnets may only
+  # be associated with a single Network ACL.
+  #
+  attribute :network_acl, kind_of: [ String, AwsNetworkAcl, AWS::EC2::NetworkACL ]
```

Comment:
From Office Hours - @dblessing suggested we could try looking up the default network ACL that subnets belong to and exchange that link with the current one when a `nil` attribute is specified.  

If it is SUPER easy to do this, go ahead.  But if its not just add a TODO explaining the issue and we can tackle it at another time.  If your use case doesn't call for it, we can wait until someone needs it.